File: Data Structures & Algorithms/search-for-word/submission-1.cpp
class Solution {
public:
    bool req(vector<vector<char>> board, string word,int pos,int i, int j){
        if (pos==word.size()){
            return true;
        }

        if(i<0 or j<0 or i>=board.size() or j>=board[0].size() or board[i][j]!=word[pos]){
            return false;
        }
        board[i][j] = '#';
        return req(board,word,pos+1,i,j+1) or req(board,word,pos+1,i-1,j) or req(board,word,pos+1,i+1,j) or req(board,word,pos+1,i,j-1);
    }
    bool exist(vector<vector<char>>& board, string word) {
        
        bool ans;
        for(int i = 0;i<board.size();i++){
            for(int j=0;j<board[0].size();j++){
                if(req(board,word,0,i,j)){
                    return true;
                }
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/generate-parentheses/submission-1.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        
        def req(l,r,curr):

            if l==0 and r==0:
                ans.append(curr)
                return
            if l<0 or r<0:
                return
            if l==r:
                # curr = curr+"("
                req(l-1,r,curr+"(")
            
            else:
                req(l-1,r,curr+"(")
                req(l,r-1,curr+")")


        ans = []
        curr = ""
        l = n
        r = n
        req(l,r,curr)
        return ans
